app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/components/TransferAdvancedSettings.kt

```diff
@@ -38,21 +38,21 @@ import com.infomaniak.swisstransfer.ui.images.icons.SpeechBubble
 import com.infomaniak.swisstransfer.ui.screen.main.settings.DownloadLimitOption
 import com.infomaniak.swisstransfer.ui.screen.main.settings.EmailLanguageOption
 import com.infomaniak.swisstransfer.ui.screen.main.settings.ValidityPeriodOption
+import com.infomaniak.swisstransfer.ui.screen.main.settings.components.SettingOption
 import com.infomaniak.swisstransfer.ui.screen.newtransfer.importfiles.PasswordTransferOption
-import com.infomaniak.swisstransfer.ui.screen.newtransfer.importfiles.TransferAdvancedOptionsEnum
 import com.infomaniak.swisstransfer.ui.theme.Margin
 import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
 
 @Composable
 fun TransferAdvancedSettings(
     modifier: Modifier = Modifier,
-    states: () -> List<TransferAdvancedOptionsEnum>,
-    onClick: () -> Unit,
+    states: () -> List<SettingOption?>,
+    onClick: (TransferAdvancedSettingType) -> Unit,
 ) {
     SwissTransferCard(modifier = modifier) {
         TransferAdvancedSettingType.entries.forEach { settingType ->
             val title by remember { derivedStateOf { states()[settingType.ordinal] } }
```

Comment:
Relying on the list being correctly order feels kind of janky. Is there no alternative?

Reply:
We can have a when on the setting type

Reply:
Fixed together with the use of states instead of initialValues